apis/core/v1alpha1/flagsourceconfiguration_types.go

```diff
@@ -92,60 +93,115 @@ type FlagSourceConfigurationSpec struct {
 	// DefaultSyncProvider defines the default sync provider
 	// +optional
 	DefaultSyncProvider SyncProviderType `json:"defaultSyncProvider"`
+
+	// SyncProviders define the syncProviders and associated configuration to be applied to the sidecar
+	// +optional
+	SyncProviders []SyncProvider `json:"syncProviders"`
+
+	// EnvVars define the env vars to be applied to the sidecar, any env vars in FeatureFlagConfiguration CRs
+	// are added at the lowest index, all values will have the EnvVarPrefix applied
+	// +optional
+	EnvVars []corev1.EnvVar `json:"envVars"`
+
+	// EnvVarPrefix defines the prefix to be applied to all environment variables applied to the sidecar, default FLAGD
+	EnvVarPrefix string `json:"envVarPrefix"`
+}
+
+type SyncProvider struct {
+	Source   string           `json:"source"`
+	Provider SyncProviderType `json:"provider"`
```

Comment:
Is there value in making the `SyncProviderType` optional and falling back to the default if it's not set?

The [`syncProviders` field](https://github.com/open-feature/open-feature-operator/pull/344/files#diff-b666c9521abcc0b8e6e7adf5bd7fa76ddbc67a97e44d05079c1e3b6f2eb10857R99) would still have a use in aggregating multiple sources to a single `FlagSourceConfiguration`. However, if we make this change, I think `sources` might be a better name for the field.

Reply:
In my opinion the `DefaultSyncProvider` configuration presented more value in versions `v1alpha1` and `v1alpha2` where the configuration was a little more convoluted, this being said I am not against reintroducing this field.
I would be keen to introduce this change in a second PR however, as this would require updates to the validation webhook for the `FlagSourceConfugration` CR (ensure that a default sync provider is set on the resource if any of the SyncProviders / Sources are missing the provider field), and I am conscious this PR is already bloated.

Reply:
I think simplifying the process to allow users to only supply sources (with the sync provider falling back to the default) makes sense if the behaviour is well documented.
I agree that renaming to `sources` makes more sense if this approach is taken.
In which case, it makes sense to do it in this release while the behaviour is already being altered and before the naming is final.

Reply:
I agree this can be done later if at all.

Reply:
Ive introduced these changes following Skyes justification - once the changes re introduced we would need a new version to change the name!